src/main/pages/developer-guide/databases.adoc

```diff
@@ -0,0 +1,177 @@
+tags: [dev-docs]
+sidebar: user_sidebar
+permalink: database.html
+folder: developer-guide
+---
+[id="currently-supported-dbs"]
+== Currently Supported Databases
+As of version 6.10, Eclipse Che currently supports following databases:
+
+* H2 (version 1.4.196)
+* PostgreSQL (version 9.6)
+
+H2 database is used by default in Single-User Che, whereas postgresql is used for Multi-user installation.
+Both of these have been tested with Selenium Integration tests, as well as TCK integration tests.
+
+It is possible to connect own DB to datasource, and have TCK tests run on it, to ensure that DB can be used with Che's JPA code.
+
+[id="datasource-components"]
+== Datasource components
+*JDBC Pool configuration*
+Depending on Database used by Che, `JNDIDataSourceFactory` implementation will provide all the required JDBC pool connection configuration.
+These properties can be overriden with Che configuration Environment Variables:
+
+[width="100%",options="header",]
+|===
+|Che Environment Variable |JDBC Pool configuration property
+|CHE_JDBC_USERNAME |username
+|CHE_JDBC_PASSWORD	|password
+|CHE_JDBC_URL	|url
+|CHE_JDBC_DRIVER\__CLASS__NAME	|driverClassName
+|CHE_JDBC_MAX__TOTAL	|maxTotal
+|CHE_JDBC_MAX__IDLE	|maxIdle
+|CHE_JDBC_MAX\__WAIT__MILLIS| maxWaitMillis
+|===
+
+(More about JDBC configuration properties at https://tomcat.apache.org/tomcat-8.5-doc/jndi-datasource-examples-howto.html)
+
+*ExceptionHandler*
+Database may sometimes produce different SQL exception, depending on the vendor.
+As in Java code, and TCK it is expected, a custom ExceptionHandler implementaion may be created, that will allow to rethrow more common exception.
+For example, in H2 and PostgreSQL, an exception may occur, when data couldn't be updated/stored due to unique constrain violation.
+But as it will a vendor specific exception with different code, there are Custom `H2ExceptionHandler`/`PostgreSqlExceptionHandler` that will catch it, and throw instead a `DuplicateKeyException`, which is used across the codebase.
+
+Exception handler has to be defined in persistence XML.
+
+[id="test-compatibility-kit"]
+== Test Compatibility Kit (TCK)
+Test Compatibility kit is a set of integration tests that covers uses of essential DAO components, that are being run against real database.
+
+List of Che modules that have TCK tests:
+
+* che-core-api-user
+* che-core-api-account
+* che-core-api-installer
+* che-core-api-ssh
+* che-core-api-workspace
+* che-core-api-workspace-activity
+* infrastructure-kubernetes
+* che-multiuser-machine-authentication (Multi User Che)
+* che-multiuser-api-organization (Multi User Che)
+* che-multiuser-api-permission (Multi User Che)
+* che-multiuser-api-resource (Multi User Che)
+* che-multiuser-permission-workspace (Multi User Che)
+
+Tests in these modules are executed on H2 Database during the build.
+Also, tests themselves are packaged into separate artifact with `tests` classifier, which are meant to be used for their re-usage against other DBs.
```

Comment:
Actually is one `artifact with 'tests' classifier` for each TCK tests.
Maybe it makes sense to make it clear, maybe not. Up to you

Reply:
rephrased, thanks